src/cli.rs

```diff
@@ -296,6 +303,11 @@ impl Cli {
         match Self::from_iter_safe(iter) {
             Ok(cli) => cli,
             Err(err) if err.kind == ErrorKind::HelpDisplayed => {
+                let mut app = if env::var("NO_COLOR").is_ok() {
+                    Cli::clap().setting(AppSettings::ColorNever)
+                } else {
+                    Cli::clap()
+                };
```

Comment:
Can we move this logic to the `clap()` function defined on L441? That would allow us to also suppress coloring in clap's `MissingRequiredArgument` error.